File: Code/C++/kth_minimum_in_array.cpp
/*
This is c++ code for finding Kth minimum element in given array
Algorithm :-
Example: Given Array=[4,1,5,7,2]
1. If given k is greater then number of elements return -1;
2. sort the given array in decreasing order(Array=[7,5,4,2,1])
3. return (k-1)th element in the given array.
*/

#include <bits/stdc++.h>
using namespace std;
int Kth_min_element(int arr[],int n,int k)
{
    if(k>n)return -1; //If given k is greater then number of elements return -1
    sort(arr,arr+n,greater<int>()); //sort the given array in decreasing order
    return arr[k-1]; //return (k-1)th element in the given array.
}
int main()
{
    int n; // Number of elements in the given array
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++)
    {
        cin>>arr[i]; // Input the elements
    }
    int k;
    cin>>k; // Input K
    cout<<Kth_min_element(arr,n,k)<<"\n";
}
/*
Test Case 1
Input :
10
3 1 6 5 8 2 98 5 67 32
4

Output :
8


Test case 2
input:
8
1 4 2 7 3 9 10 8
4
Output:
7

Time-Complexity : O(nlogn)
Space-Complexity : O(n)
*/

File: Code/Python/Kth_Minimum_in_array.py
# Python3 program to find k'th smallest element
 
# Function to return k'th smallest element in a given array
def kthSmallest(arr, n, k):
 
    # Sort the given array
    arr.sort()
 
    # Return k'th element in the
    return arr[k-1]
 
# Driver code
if __name__=='__main__':
    arr = []
    n = int(input("Enter number of elements: "))
    print("Enter elements: ")
    for i in range(n):
        element = int(input())
        arr.append(element)

    k = int(input("Enter the number k: "))
    print("K'th smallest element is",
          kthSmallest(arr, n, k))

'''
Time Complexity: O(N Log N)
Space Complexity: O(N)

Example 1:
Input:
Enter number of elements: 
6
Enter elements: 
7 10 4 3 20 15

Enter the number k: 
3

Output:
K'th smallest element is:
7

Example 2:
Input:
Enter number of elements: 
7
Enter elements: 
8 1 0 9 5 11 4

Enter the number k: 
4

Output:
K'th smallest element is:
5
'''
